packages/core/src/public.ts

```diff
@@ -0,0 +1,37 @@
+import type { DefinitionContentType } from '@glazed/did-datastore'
+
+import type { ModelTypes as CoreModelTypes } from './__generated__/model'
+import type { Core } from './core'
+
+export type PublicIDParams<ModelTypes extends CoreModelTypes = CoreModelTypes> = {
+  core: Core<ModelTypes>
+  id: string
+}
+
+/**
+ * ```sh
+ * import { PublicID } from '@self.id/core'
+ * ```
+ */
+export class PublicID<
+  ModelTypes extends CoreModelTypes = CoreModelTypes,
+  Alias extends keyof ModelTypes['definitions'] = keyof ModelTypes['definitions']
+> {
+  #core: Core<ModelTypes, Alias>
```

Comment:
Can you remind me about that `#` does? :)

Reply:
It makes it a private field, as we target ES2020 we can use these new APIs :)